pandas/io/json/_normalize.py

```diff
@@ -261,9 +261,12 @@ def _pull_records(js: Dict[str, Any], spec: Union[List, str]) -> List:
 
     if isinstance(data, list) and not data:
         return DataFrame()
-
-    # A bit of a hackjob
-    if isinstance(data, dict):
+    elif isinstance(data, abc.Iterator):
+        # GH35923 Fix pd.json_normalize to not skip the first element of a
+        # generator input
+        data = list(data)
```

Comment:
This could have some big performance implications when dealing with large generators - is it not alternately possible to just store the first element for inspection and reuse as necessary while maintaining the state of the generator?

Reply:
we *barely* support generators (its not even documented), so -1 if this adds *any* complexity.